src/careamics/dataset/tiling/lvae_tiled_patching.py

```diff
@@ -93,19 +96,98 @@ def extract_tiles(
             yield tile, tile_info
 
 
+def compute_tile_info_legacy(
+    grid_index_manager: GridIndexManager, index: int
+) -> TileInformation:
+    """
+    Compute the tile information for a tile at a given dataset index.
+
+    Parameters
+    ----------
+    grid_index_manager : GridIndexManager
+        The grid index manager that keeps track of tile locations.
+    index : int
+        The dataset index.
+
+    Returns
+    -------
+    TileInformation
+        Information that describes how to crop and stitch a tile to create a full image.
+
+    Raises
+    ------
+    ValueError
+        If `grid_index_manager.data_shape` does not have 4 or 5 dimensions.
+    """
+    data_shape = np.array(grid_index_manager.data_shape)
+    if len(data_shape) == 5:
+        n_spatial_dims = 3
+    elif len(data_shape) == 4:
+        n_spatial_dims = 2
+    else:
+        raise ValueError("Data shape must have 4 or 5 dimensions, equating to SC(Z)YX.")
+
+    stitch_coords_start = np.array(
+        grid_index_manager.get_location_from_dataset_idx(index)
+    )
+    stitch_coords_end = stitch_coords_start + np.array(grid_index_manager.grid_shape)
+
+    tile_coords_start = stitch_coords_start - grid_index_manager.patch_offset()
+
+    # --- replace out of bounds indices
+    out_of_lower_bound = stitch_coords_start < 0
+    out_of_upper_bound = stitch_coords_end > data_shape
+    stitch_coords_start[out_of_lower_bound] = 0
+    stitch_coords_end[out_of_upper_bound] = data_shape[out_of_upper_bound]
+
+    # TODO: TilingMode not in current version
+    # if grid_index_manager.tiling_mode == TilingMode.ShiftBoundary:
```

Comment:
I see this as the default tiling mode. this is what you have in CAREamics: shifting the boundary tile such that you get prediction for the full frame without using any padding.